examples/typescript/pipe-screen-time-storyteller/pipe.ts

```diff
@@ -0,0 +1,357 @@
+const INTERVAL = 60 * 60 * 1000; // 1 hour in milliseconds
+
+interface AIProvider {
+    provider: string;
+    model: string;
+}
+
+interface Config {
+    aiProvider: string;
+    models: {
+        claude: string;
+        openai: string;
+        ollama: string;
+    };
+    apiKeys: {
+        claude: string;
+        openai: string;
+    };
+    pageSize: number;
+    contentType: ContentType;
+    github: {
+        personalAccessToken: string;
+    };
+}
+
+interface NarrativeSummary {
+    summary: string;
+    mood: string; // This will now be an emoji
+    keyInsights: string[]; // Each insight will include an emoji
+}
+
+type ContentType = "ocr" | "audio" | "all"; // Add this near the top of your file
+
+interface ContentItem {
+    // Define the structure of your content items here
+    // For example:
+    id: string;
+    content: string;
+    timestamp: string;
+    // Add other relevant fields
+}
+
+async function getAIProvider(config: Config): Promise<AIProvider> {
+    return { provider: config.aiProvider, model: config.models[config.aiProvider] };
+}
+
+async function queryScreenpipe(params: ScreenpipeQueryParams): Promise<ScreenpipeResponse | null> {
```

Comment:
you don't need this

use [pipe.queryScreenpipe](https://github.com/mediar-ai/screenpipe/blob/e4d26ed9ea59328f092705f5b7ddb5b2ce49a936/screenpipe-core/src/deno/runtime.js#L169)

Reply:
Okay.